backend/wmg/api/v1.py

```diff
@@ -194,26 +189,34 @@ def agg_cell_type_counts(cell_counts: DataFrame) -> DataFrame:
     cell_counts_cell_type_agg = cell_counts.groupby(
         ["tissue_ontology_term_id", "cell_type_ontology_term_id"], as_index=True
     ).sum()
-    cell_counts_cell_type_agg.rename(
-        columns={"n_total_cells": "n_cells_cell_type", "n_total_cells_rollup": "n_cells_cell_type_rollup"}, inplace=True
-    )
+    cell_counts_cell_type_agg.rename(columns={"n_total_cells": "n_cells_cell_type"}, inplace=True)
     return cell_counts_cell_type_agg
 
 
 def agg_tissue_counts(cell_counts: DataFrame) -> DataFrame:
     # Aggregate cube data by tissue
     cell_counts_tissue_agg = cell_counts.groupby(["tissue_ontology_term_id"], as_index=True).sum()
-    cell_counts_tissue_agg.rename(
-        columns={"n_total_cells": "n_cells_tissue", "n_total_cells_rollup": "n_cells_tissue_rollup"}, inplace=True
-    )
+    cell_counts_tissue_agg.rename(columns={"n_total_cells": "n_cells_tissue"}, inplace=True)
     return cell_counts_tissue_agg
 
 
-def get_dot_plot_data(query_result: DataFrame, cell_counts: DataFrame) -> Tuple[DataFrame, DataFrame]:
+def get_dot_plot_data(query_result: DataFrame, cell_counts: DataFrame, is_rollup: bool) -> Tuple[DataFrame, DataFrame]:
     # Get the dot plot matrix dataframe and aggregated cell counts per cell type
     cell_counts_cell_type_agg = agg_cell_type_counts(cell_counts)
     cell_counts_tissue_agg = agg_tissue_counts(cell_counts)
     dot_plot_matrix_df = build_dot_plot_matrix(query_result, cell_counts_cell_type_agg, cell_counts_tissue_agg)
+
+    if is_rollup:
```

Comment:
nit: to keep this method simple, maybe move this conditional block to its own method and then have the caller do:
```
dot_plot_matrix_df, cell_counts_cell_type_agg = get_dot_plot_data(...)
if rollup:
    dot_plot_matrix_df, cell_counts_cell_type_agg = rollup(dot_plot_matrix_df, cell_counts_cell_type_agg)
```
maybe this helps simplify tests as well